File: java/src/main/java/br/datastructures/stack/AbstractStack.java
package br.datastructures.stack;

public abstract class AbstractStack<I> implements Stack<I> {
    protected int quantity;

    public AbstractStack() {
        quantity = 0;
    }

    @Override
    public boolean empty() {
        return size() == 0;
    }
    
    @Override
    public int size() {
        return quantity;
    }
}

File: csharp/stack/AbstractStack.cs
namespace CSharp.Stack;

using CSharp.Stack;

public abstract class AbstractStack<T> : IMyStack<T>
{
    protected int Quantity;

    public AbstractStack()
    {
        Quantity = 0;
    }

    public bool IsEmpty()
    {
        return Quantity == 0;
    }

    public int Size()
    {
        return Quantity;
    }

    public abstract void Push(T item);
    public abstract T Pop();
    public abstract T GetTop();
}